File: ruby/g00.inheritance/Student.rb
#!/usr/bin/ruby
# class declaration
class Student < Person
  ###### Mutators ######
  def setGPA (gpa)
    @gpa = gpa         # set instance data
  end
  
  ###### Accessors ######
  def getGPA
    @gpa               # return instance data
  end
end

File: js/ts/g00.inheritance/Student.ts
///<reference path='Person.ts' />
/******* RHINO/SPIDERMONKEY SUPPORT *******/
declare function load(str);
load('Person.js');
/******* CLASS DECLARATION *******/
class Student extends Person {
  private _gpa: number;

  /******* PROPERTIES *******/
  get gpa(): number { return this._gpa; }   // getter
  set gpa(gpa: number) { this._gpa = gpa; } // setter
}
